hypertension/dc-7101-algorithms/py-root/test/main_test.py

```diff
@@ -1,14 +1,214 @@
+import pytest
 from lib.main import main
 
-def test_main():
-    test_event = {"irrelevant": "value"}
-    assert main(test_event) == {
-        "statusCode": 200,
-        "headers": {
-            "Access-Control-Allow-Headers" : "Content-Type",
-            "Access-Control-Allow-Origin": "*",
-            "Access-Control-Allow-Methods": "OPTIONS,POST"
-        },
-        "body": "test body"
-    }
+@pytest.mark.parametrize(
+    "request_data, response",
+    [
+        # All three calculator functions return valid results readings
+        (
+            {
+                "bp": [
+                    {
+                        "diastolic": 115,
+                        "systolic": 180,
+                        "date": "2021-11-01"
+                    },
+                    {
+                        "diastolic": 110,
+                        "systolic": 200,
+                        "date": "2021-09-01"
+                    }
+                ],
+                "medication": ["Capoten"],
+                'date_of_claim': '2021-11-09',
+                'veteran_is_service_connected': True
+            },
+            {
+                "statusCode": 200,
+                "headers": {
+                    "Access-Control-Allow-Headers" : "Content-Type",
+                    "Access-Control-Allow-Origin": "*",
+                    "Access-Control-Allow-Methods": "OPTIONS,POST"
+                },
+                "body": {
+                    "predominance_calculation": {
+                        "success": True,
+                        'predominant_diastolic_reading': 115,
+                        'predominant_systolic_reading': 180
```

Comment:
`predominant_systolic_reading` should be 200, correct?

Reply:
Indeed. This is a bug in the algorithm we believe.

Created https://amida.atlassian.net/browse/MCP-1049 to fix it.

(Resolved)